File: Libs/pollilib/javascript/tests/test_stt_vision.js
import test from 'node:test';
import assert from 'node:assert/strict';
import fs from 'node:fs';
import path from 'node:path';

import { PolliClient } from '../polliLib/index.js';
import { FakeResponse, SeqFetch } from './helpers.js';

test('transcribe_audio from tmp file', async () => {
  const tmpDir = await fs.promises.mkdtemp(path.join(process.cwd(), 'tmp-stt-'));
  const audioPath = path.join(tmpDir, 'a.wav');
  await fs.promises.writeFile(audioPath, Buffer.from('RIFF....WAVEfmt '));
  const seq = new SeqFetch([ new FakeResponse({ jsonData: { choices: [ { message: { content: 'transcribed' } } ] } }) ]);
  const c = new PolliClient({ fetch: seq.fetch.bind(seq) });
  const out = await c.transcribe_audio(audioPath);
  assert.equal(out, 'transcribed');
  const body = JSON.parse(seq.calls[0].opts.body);
  assert.equal(body.safe, false);
});

test('analyze_image_url and analyze_image_file', async () => {
  const seq = new SeqFetch([ new FakeResponse({ jsonData: { choices: [ { message: { content: 'This is a bridge' } } ] } }) ]);
  const c = new PolliClient({ fetch: seq.fetch.bind(seq) });
  const out1 = await c.analyze_image_url('http://x/y.jpg');
  assert.equal(out1, 'This is a bridge');
  const visionBody1 = JSON.parse(seq.calls[0].opts.body);
  assert.equal(visionBody1.safe, false);
  // For file path, inject another response
  seq.responses.push(new FakeResponse({ jsonData: { choices: [ { message: { content: 'This is a bridge' } } ] } }));
  const tmpDir = await fs.promises.mkdtemp(path.join(process.cwd(), 'tmp-vis-'));
  const imgPath = path.join(tmpDir, 'i.jpg');
  await fs.promises.writeFile(imgPath, Buffer.from([0xFF,0xD8,0xFF]));
  const out2 = await c.analyze_image_file(imgPath);
  assert.equal(out2, 'This is a bridge');
  const visionBody2 = JSON.parse(seq.calls[1].opts.body);
  assert.equal(visionBody2.safe, false);
});

File: Libs/pollilib/python/tests/test_stt_vision.py
import os
import tempfile

from polliLib import PolliClient
from .conftest import FakeResponse, FakeSession


def test_transcribe_audio_tmpfile(tmp_path: tempfile.TemporaryDirectory):
    # Create a tiny fake wav/mp3 file (content is irrelevant for test)
    audio_path = os.path.join(tmp_path, 'sample.wav')
    with open(audio_path, 'wb') as f:
        f.write(b'RIFF....WAVEfmt ')

    fs = FakeSession()
    fs.post = lambda url, **kw: FakeResponse(json_data={"choices": [{"message": {"content": "transcribed"}}]})
    c = PolliClient(session=fs)
    out = c.transcribe_audio(audio_path)
    assert out == 'transcribed'


def test_vision_analyze_url_and_file(tmp_path: tempfile.TemporaryDirectory):
    fs = FakeSession()
    fs.post = lambda url, **kw: FakeResponse(json_data={"choices": [{"message": {"content": "This is a bridge"}}]})
    c = PolliClient(session=fs)
    # URL
    out1 = c.analyze_image_url('http://x/y.jpg')
    assert out1 == 'This is a bridge'
    # File
    img_path = os.path.join(tmp_path, 'img.jpg')
    with open(img_path, 'wb') as f:
        f.write(b'\xff\xd8\xff')
    out2 = c.analyze_image_file(img_path)
    assert out2 == 'This is a bridge'

